Added git hook documentation for Umbraco Deploy (#4646)
umbraco/UmbracoDocs: Add-ons/Umbraco-Deploy/Installing-Deploy/Streamlining-Local-Development/index.md

---
versionFrom: 9.0.0
versionTo: 10.0.0
meta.Title: "Streamlining Local Development"
meta.Description: "Additional steps you can carry out to streamline your local development workflow"
---

# Streamlining Local Development

In this section we discuss some additional steps you can carry out to streamline your local development workflow.

## Creating Git Hooks

> When working in a project team, it's common for a developer pulling the code from source control to want to update their local environment with the latest Umbraco schema.

Reviewer: ```suggestion
Working in a team, it's common for developers to pull code from source control to update their local environment with the latest Umbraco schema.
```

They can do this by starting up the website, navigating to the _Settings > Deploy_ dashboard and triggering a data extraction.

We can automate this step using a [git hook](https://www.atlassian.com/git/tutorials/git-hooks).

When working with Umbraco Cloud, this step is configured automatically for you when you clone and run your project the first time. If working with Umbraco Deploy On-Premise, you can set it up yourself.

> The process works by making use of the marker file Umbraco Deploy uses to trigger an update of the Umbraco schema from the `.uda` files retrieved from source control.  If a file by the name of `deploy-on-start` is found in the `/umbraco/Deploy` folder, an update will be run automatically when the site starts up. As such, if we ensure that file is created everytime the source code is pulled from the remote repository, we can automate the update.

Reviewer: ```suggestion
The process works by using the marker file Umbraco Deploy uses to trigger an update of the Umbraco schema from the `.uda` files retrieved from source control.  

If a file named `deploy-on-start` is found in the `/umbraco/Deploy` folder, an update will be run automatically when the site starts up. 

Therefore, if we ensure that the file is created everytime the source code is pulled from the remote repository, we can automate the update.
```
